src/Database/Bloodhound/Types.hs

```diff
@@ -5212,3 +5213,10 @@ instance FromJSON NodeNetworkInterface where
       parse o = NodeNetworkInterface <$> o .: "mac_address"
                                      <*> o .: "name"
                                      <*> o .: "address"
+
+newtype MaybeNA a = MaybeNA { unMaybeNA :: Maybe a }
+  deriving (Show, Eq)
+
+instance FromJSON a => FromJSON (MaybeNA a) where
+  parseJSON (String "NA") = pure $ MaybeNA Nothing
+  parseJSON o = MaybeNA . Just <$> parseJSON o
```

Comment:
That's fine then. I don't like the loss of provenance, but if it suits you, I won't object.